openstack/mysql/install.sls

```diff
@@ -117,6 +118,12 @@ root-localhost-test:
     - name: mysql -u root -ppassword -e "ALTER USER 'root'@'localhost' IDENTIFIED BY '{{virl.mypassword}}';"
     - unless: mysql -u root -p{{virl.mypassword}} -e quit
 
+root-password-change:
+{% if old_mypassword != '' and old_mypassword != virl.mypassword %}
```

Comment:
The conditional has to be outside of the state id.  Otherwise, if it evaluates false the rest of the id is null and the state cannot be rendered (no dictionary).